M2/Macaulay2/m2/set.m2

```diff
@@ -83,8 +83,14 @@ product VirtualTally := (w) -> product(pairs w, (k,v) -> k^v)
 Set.synonym = "set"
 
 -- constructors, both compiled functions defined in d/sets.dd
-set HashTable := set VisibleList := Set => set
-new Set from List := new Set from HashTable := Set => (X,x) -> set x
+set' = set
+set = method(TypicalValue => Set, Dispatch => Thing)
+set Set         := identity
+set HashTable   :=
+set VisibleList := Set => set'
+new Set from Set         := Set => (Set, S) -> set' S
```

Comment:
What's the point of this line? isn't that covered by `new Set from HashTable`?

Reply:
It seems appropriate given the three lines above, even if just for emphasis and documentation.

Reply:
I often trip over `Set set`, in that I am making several things a set, and some already are, some are just lists.  I kind of like this addition.